lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -43,30 +42,35 @@ class TraceOfTab extends ComputedArtifact {
   */
   compute_(trace) {
     // Parse the trace for our key events and sort them by timestamp.
-    const keyEvents = trace.traceEvents.filter(e => {
-      return e.cat.includes('blink.user_timing') || e.cat.includes('loading') ||
+    const keyEvents = trace.traceEvents
+      .filter(e => {
+        return e.cat.includes('blink.user_timing') ||
+          e.cat.includes('loading') ||
           e.name === 'TracingStartedInPage';
-    }).sort((event0, event1) => event0.ts - event1.ts);
+      })
+      .sort((event0, event1) => event0.ts - event1.ts);
 
     // The first TracingStartedInPage in the trace is definitely our renderer thread of interest
     // Beware: the tracingStartedInPage event can appear slightly after a navigationStart
     const startedInPageEvt = keyEvents.find(e => e.name === 'TracingStartedInPage');
     // Filter to just events matching the frame ID for sanity
     const frameEvents = keyEvents.filter(e => e.args.frame === startedInPageEvt.args.data.page);
 
+    // Our navStart will be the last frame navigation in the trace
+    const navigationStart = frameEvents.filter(e => e.name === 'navigationStart').pop();
+
     // Find our first paint of this frame
-    const firstPaint = frameEvents.find(e => e.name === 'firstPaint');
-    // Our navStart will be the latest one before fP.
-    const navigationStart = frameEvents.filter(e =>
-        e.name === 'navigationStart' && e.ts < firstPaint.ts).pop();
+    const firstPaint = frameEvents.find(e => e.name === 'firstPaint' && e.ts > navigationStart.ts);
 
     // FCP will follow at/after the FP
-    const firstContentfulPaint = frameEvents.find(e =>
-      e.name === 'firstContentfulPaint' && e.ts >= firstPaint.ts);
+    const firstContentfulPaint = frameEvents.find(
+      e => e.name === 'firstContentfulPaint' && e.ts > navigationStart.ts
```

Comment:
we were using FP as the reference event before. Honestly this line could been asking for > navStart rather than FP, but it was just all FP for consistency.